packages/dma-library/src/strategies/aave/common/get-flashloan-token.ts

```diff
@@ -1,15 +1,32 @@
 import { AaveLikeStrategyAddresses } from '@dma-library/operations/aave-like'
-import { WithFlashloanToken } from '@dma-library/types'
+import { WithFlashLoanArgs } from '@dma-library/types'
+import { AaveLikeProtocol } from '@dma-library/types/protocol'
 
 export interface FlashloanDependencies {
+  protocol: AaveLikeProtocol
   network: string
   addresses: AaveLikeStrategyAddresses
+  debt: {
+    symbol: string
+    address: string
+    precision: number
+  }
 }
 
 export function getFlashloanToken({
   network,
   addresses,
-}: FlashloanDependencies): WithFlashloanToken {
+  protocol,
+  debt,
+}: FlashloanDependencies): WithFlashLoanArgs {
+  if (protocol === 'Spark') {
```

Comment:
Is it possible to use an `enum` here instead of a harcoded string?

Reply:
The protocol variable is of type: `export type AaveLikeProtocol = 'AAVE' | 'AAVE_V3' | 'Spark'`, so it's not hardcoded per se. ;)